### ISSUE TITLE
Database Design

### ISSUE THREAD

**Issue by benjamesfleming** [benjamesfleming/shopcheap#2]
To get this website up and running we need to get our database design worked out. 

The main tables include:

- `Users`
- `Recipes`
- `Ingredients`
- `Shops`
- `IngredientPriceHistory`

Other tables:

- `RecipeIngredients`
- `Friends`
- `Following`

The main tables are all self-explanatory, but the following tables are explained below.

**IngredientPriceHistory**:
This table hold hold extra price history for all the ingredients. This table compliments the main Ingredients table. The only columns should be: `ingredient_id`, `date`, `price`.

**RecipeIngredients**:
This table holds the relationship between the ingredient and recipe tables.

**Friends**:
This table holds the relationship between between users. one user can be friends with many users.

**Following**:
This table holds the relationship between users and recipes. one user can follow many recipes. when a user follows a recipe it will show up in that users list of recipes. but they will not be able to edit it. maybe we can think about the ability to fork recipes.

Any suggestions are welcome.

**Comment by RobertMadx** [comment]
Agree, think we can add later if we need to. Getting this to work will be an achievement in itself. I don't feel confident on how to manage Ingredients vs recipes and think once we start using/testing it we might get a better idea of how it should work. Like how do we categorise stuff, Milk can be trim, standard, etc. Maybe have them as separate ingredients, although we might need to link them somehow. For example, a recipe might be able to use ingredient A or B or C(this might be a thing in the recipe)... OR maybe a grouping of sorts on the ingredient....

**Comment by benjamesfleming** [comment]
Milk and trim milk should be separate items with separate prices and histories. Other things like 100g bag of chips vs 200g bag of chips could be determined by one ingredient. Maybe a recipe uses 100g of chips but the price is calculated by the price of the 200g bag / 2. This means we would need to keep track of the quanties in the ingredients table, along side the costs.
